src/main/java/org/prebid/server/bidder/gotthamads/GothamAdsBidder.java

```diff
@@ -0,0 +1,142 @@
+package org.prebid.server.bidder.gotthamads;
+
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.iab.openrtb.request.BidRequest;
+import com.iab.openrtb.request.Device;
+import com.iab.openrtb.request.Imp;
+import com.iab.openrtb.response.Bid;
+import com.iab.openrtb.response.BidResponse;
+import io.vertx.core.MultiMap;
+import io.vertx.core.http.HttpMethod;
+import org.prebid.server.bidder.Bidder;
+import org.prebid.server.bidder.model.BidderBid;
+import org.prebid.server.bidder.model.BidderCall;
+import org.prebid.server.bidder.model.BidderError;
+import org.prebid.server.bidder.model.HttpRequest;
+import org.prebid.server.bidder.model.Result;
+import org.prebid.server.exception.PreBidException;
+import org.prebid.server.json.DecodeException;
+import org.prebid.server.json.JacksonMapper;
+import org.prebid.server.proto.openrtb.ext.ExtPrebid;
+import org.prebid.server.proto.openrtb.ext.request.gothamads.GothamAdsImpExt;
+import org.prebid.server.proto.openrtb.ext.response.BidType;
+import org.prebid.server.util.HttpUtil;
+import org.prebid.server.util.ObjectUtil;
+
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Optional;
+
+public class GothamAdsBidder implements Bidder<BidRequest> {
+
+    private static final TypeReference<ExtPrebid<?, GothamAdsImpExt>> TYPE_REFERENCE = new TypeReference<>() {
+    };
+    private static final String ACCOUNT_ID_MACRO = "{{AccountId}}";
+    private static final String X_OPENRTB_VERSION = "2.5";
+    private static final Map<Integer, BidType> ALLOWED_IMP_TYPES = Map.of(
+            1, BidType.banner,
+            2, BidType.video,
+            4, BidType.xNative
+    );
+
+    private final String endpointUrl;
+    private final JacksonMapper mapper;
+
+    public GothamAdsBidder(String endpointUrl, JacksonMapper mapper) {
+        this.endpointUrl = HttpUtil.validateUrl(Objects.requireNonNull(endpointUrl));
+        this.mapper = Objects.requireNonNull(mapper);
+    }
+
+    @Override
+    public Result<List<HttpRequest<BidRequest>>> makeHttpRequests(BidRequest request) {
+        final GothamAdsImpExt impExt;
+
+        try {
+            impExt = request.getImp().stream().findFirst()
+                    .map(this::parseImpExt)
+                    .orElseThrow(() -> new PreBidException("Any impressions were not provided."));
+        } catch (PreBidException e) {
+            return Result.withErrors(List.of(BidderError.badInput(e.getMessage())));
+        }
+
+        final HttpRequest<BidRequest> httpRequest = HttpRequest.<BidRequest>builder()
+                .method(HttpMethod.POST)
+                .uri(resolveEndpoint(impExt.getAccountId()))
+                .headers(headers(request))
+                .body(mapper.encodeToBytes(request))
```

Comment:
You are still not clearing first imps extension as in go code.

Reply:
maybe it's more convenient, but as I mentioned in another comment it requires creating new bid request due to immutable nature of the objects

Reply:
Our outgoing requests should be the same as Go(nothing should be different), so please update